### PR TITLE
guides: update guides to reflect restart_process extension instead of entr hack for restarting

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @nicks, @jazzdan,

Please review the following commits I made in branch maiamcc/update-ex-guides-restart_process:

061cf97b5ea118d27de643a7f832360c3815d77e (2020-05-15 19:22:23 -0400)
guides: update guides to reflect restart_process extension instead of entr hack for restarting

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
would love to have @ellenkorbes review this instead of me. 

How we introduce extensions here is going to speak a lot to how we introduce extensions to developers in general

Like, an alternative way to do this would be to break it up into two separate steps: "here's how you'd wire live_update with just run() calls. But the recommended way to do it is with an extension. extensions are a great way to share best practices with other people. here's how you do it"

But I could be convinced that this is too much for an initial best practices guide

File: docs/example_csharp.md [hunk 1]
```diff
@@ -206,23 +206,28 @@ With Tilt, we can skip all of these steps, and instead [live_update](live_update
...
-We've added a `live_update` parameter to `docker_build()` with a `sync` step. This copies the build output from the `out` directory into container.
+The first thing to notice is the `live_update` parameter, which consists of one `sync` step.
+This copies the build output from the `out` directory into the container.
 
-We've also added a new parameter: `entrypoint='find . | entr -r dotnet hello-tilt.dll'`.
-
-`entr` is a tool that automatically restarts a shell command whenever the watched
-file changes. This command restarts our server every time the pod is updated.
+After syncing the files, we want to re-execute our updated binary. We accomplish this with the
+[`restart_process` extension](https://github.com/windmilleng/tilt-extensions/tree/master/restart_process)
```
Comment (ellenkorbes) [inline docs/example_csharp.md:226]: SGTM, I'd just add a short remark about what the `load(ext)` call is. E.g.

```suggestion
[`restart_process` extension](https://github.com/windmilleng/tilt-extensions/tree/master/restart_process), which we imported on the first line
```

Comment (maiamcc) [reply docs/example_csharp.md:226]: ooh good call, ty

**Comment by maiamcc** [comment]
@ellenkorbes (or whoever) can I get a ✅ ?

**Comment by ellenkorbes** [comment]
There!

**Pull request merged**